File: Easy/Python/1394. Find Lucky Integer in an Array.py
class Solution(object):
    def findLucky(self, arr):
        arr.sort()
        count = [0] * (arr[len(arr) - 1] + 1)
        largest = -1

        for n in arr:
            count[n] += 1
        for i in range(1, len(count)):
            if count[i] == i:
                largest = i
        
        return largest
        

File: Easy/Java/1394. Find Lucky Integer in an Array.java
class Solution {
    public int findLucky(int[] arr) {
        Arrays.sort(arr);
        int[] count = new int[arr[arr.length - 1] + 1];
        int largest = -1;
        
        for(int n : arr)
            count[n]++;
        for(int i = 1; i < count.length; i++){
            if(count[i] == i)
                largest = i;
        }

        return largest;
    }
}
